src/test/kotlin/uk/gov/dluhc/printapi/service/PrintRequestsServiceTest.kt

```diff
@@ -39,31 +48,44 @@ class PrintRequestsServiceTest {
     @Captor
     private lateinit var savedRecordArgumentCaptor: ArgumentCaptor<PrintDetails>
 
+    @Captor
+    private lateinit var savedCertificateArgumentCaptor: ArgumentCaptor<Certificate>
+
     @Test
     fun `should batch multiple print requests, save and submit to queue`() {
         // Given
         val batchSize = 5
         val numOfRequests = 12
         val items = (1..numOfRequests).map { buildPrintDetails() }
+        val certificates = (1..numOfRequests).map { certificateBuilder(status = PENDING_ASSIGNMENT_TO_BATCH) }
 
         val batchId1 = aValidBatchId()
         val batchId2 = aValidBatchId()
         val batchId3 = aValidBatchId()
+        val batchId4 = aValidBatchId()
+        val batchId5 = aValidBatchId()
+        val batchId6 = aValidBatchId()
 
-        given(printDetailsRepository.getAllByStatus(Status.PENDING_ASSIGNMENT_TO_BATCH)).willReturn(items)
-        given(idFactory.batchId()).willReturn(batchId1, batchId2, batchId3)
+        given(printDetailsRepository.getAllByStatus(PENDING_ASSIGNMENT_TO_BATCH)).willReturn(items)
+        given(certificateRepository.findByStatusIs(PENDING_ASSIGNMENT_TO_BATCH)).willReturn(certificates)
+        given(idFactory.batchId()).willReturn(batchId1, batchId2, batchId3, batchId4, batchId5, batchId6)
 
         // When
         printRequestsService.processPrintRequests(batchSize)
 
         // Then
-        verify(printDetailsRepository).getAllByStatus(Status.PENDING_ASSIGNMENT_TO_BATCH)
-        verify(idFactory, times(3)).batchId()
+        verify(printDetailsRepository).getAllByStatus(PENDING_ASSIGNMENT_TO_BATCH)
+        verify(certificateRepository).findByStatusIs(PENDING_ASSIGNMENT_TO_BATCH)
+        verify(idFactory, times(3 * 2)).batchId()
         verify(printDetailsRepository, times(12)).save(any())
-        verify(processPrintRequestQueue, times(3)).submit(any())
+        verify(certificateRepository, times(12)).save(any())
+        verify(processPrintRequestQueue, times(3 * 2)).submit(any())
```

Comment:
Done as `*2` just to highlight which will need the `*2` removed when Dynamo is removed